File: DNA.VisualBasic/DNA.VisualBasic/Helpers/HelperMethods.vb
Module HelperMethods
    Async Function IsWPF(context As CodeRefactorings.CodeRefactoringContext) As Task(Of Boolean)
        Dim model = Await context.Document.GetSemanticModelAsync
        Dim isSupported = model.Compilation.GetTypeByMetadataName("System.Windows.Navigation.JournalEntry")

        If isSupported Is Nothing Then
            Return False
        Else
            Return True
        End If
    End Function
End Module

File: DNA.CSharp/DNA.CSharp/Helpers/HelperMethods.cs
using System.Threading.Tasks;
using Microsoft.CodeAnalysis.CodeRefactorings;

namespace DNA.CSharp.Helpers
{
    static class HelperMethods
    {
        async static Task<bool> IsWPF(CodeRefactoringContext context)
        {
            var model = await context.Document.GetSemanticModelAsync();
            var isSupported = model.Compilation.GetTypeByMetadataName("System.Windows.Navigation.JournalEntry");

            if (isSupported==null)
            {
                return false;
            }
            else
            {
                return true;
            }
        }
    }
}
